google/cloud/bigquery/client.py

```diff
@@ -44,6 +45,13 @@
 import uuid
 import warnings
 
+try:
+    import pyarrow  # type: ignore
+
+    _PYARROW_VERSION = packaging.version.parse(pyarrow.__version__)
```

Comment:
Nit: Whereever we're using this, let's move that logic to PyarrowVersions, instead.

Reply:
Filed https://github.com/googleapis/python-bigquery/issues/1430